src/state/nftMarket/helpers.ts

```diff
@@ -59,7 +65,87 @@ export const getNftsMarketData = async (collectionAddress: string): Promise<Nfts
     )
     return res.collection.NFTs
   } catch (error) {
-    console.error('Failed to fetch NFTs', error)
+    console.error('Failed to fetch NFTs from collection', error)
+    return []
+  }
+}
+
+export const getNftsMarketData = async (where = {}): Promise<NftSubgraphEntity[]> => {
+  try {
+    const res = await request(
+      GRAPH_API_NFTMARKET,
+      gql`
+        query getNftsMarketData($where: NFT_filter) {
+          nfts(where: $where) {
+            tokenId
+            metadataUrl
+            transactionHistory {
+              id
+              block
+              timestamp
+              askPrice
+              netPrice
+              buyer {
+                id
+              }
+              seller {
+                id
+              }
+              withBNB
+            }
+          }
+        }
+      `,
+      { where },
+    )
+
+    return res.nfts
+  } catch (error) {
+    console.error('Failed to fetch NFTs market data', error)
     return []
   }
 }
+
+export const fetchWalletNftIds = async (account: string): Promise<string[]> => {
+  const nftIdPromises = Object.keys(collections).map(async (nftCollectionKey) => {
```

Comment:
use lodash map instead

Reply:
so you can remove line 116 because the object will be passed as a param. 
For the 100th time, the config file won't stay so don't add dependencies on it.

Reply:
Yea I know that Doge - 

- I needed to throw something together to query user wallet NFTs for an NFT collection address. The important bit isn't accessing the collection's contract address (the config dependency) - it's how we use that address to find a users' NFTs
- This temporary and is reusing existing code that exists in the code base (i.e. we would always need to decouple this regardless of whether I reused it here or not). 
- The only thing the collections config is used for here is the contract address - it ought to be straightforward to decouple once that data is being stored in state.

Reply:
Now that the collections API is returning data I'll migrate this 👍

Reply:
Now receives a `collections` array, retrieved from state